### PR TITLE
ESRI web services

### PR DESCRIPTION
**Pull request by eliotjordan**
Adds support for ArcGIS REST API services. The original PR (#335) was closed in order to reboot the discussion and to facilitate moving the branch into the main repository.

### CODE REVIEW

**Comment by krdyke** [comment]
Here's a [gist](https://gist.github.com/krdyke/b9f5a71f0a46f3cb8bec) with some records. Unfortunately we don't have a tiled layer that uses Web Mercator. Let me know what needs fixing/tweaking.

**Comment by eliotjordan** [comment]
@krdyke Thanks! I'm adding the docs now.

File: lib/geoblacklight/constants.rb [hunk 1]
```diff
@@ -14,7 +14,11 @@ module Constants
       wcs: 'http://www.opengis.net/def/serviceType/ogc/wcs',
       wfs: 'http://www.opengis.net/def/serviceType/ogc/wfs',
       wms: 'http://www.opengis.net/def/serviceType/ogc/wms',
-      hgl: 'http://schema.org/DownloadAction'
+      hgl: 'http://schema.org/DownloadAction',
+      feature_layer: 'http://www.arcgis.com/rdf#FeatureLayer',
```
Comment (mejackreed) [inline lib/geoblacklight/constants.rb]: Can you remind me why we decided on RDF here? To me referencing "RDF" feels like it should actually be rdf, no?

**Comment by eliotjordan** [comment]
The ESRI service refs in dct_references now use the base [ArcGIS Service uri from Cat-Interop](https://github.com/OSGeo/Cat-Interop/blob/master/LinkPropertyLookupTable.csv#L24).

      feature_layer: 'urn:x-esri:serviceType:ArcGIS#FeatureLayer',
      tiled_map_layer: 'urn:x-esri:serviceType:ArcGIS#TiledMapLayer',
      dynamic_map_layer: 'urn:x-esri:serviceType:ArcGIS#DynamicMapLayer',
      image_map_layer: 'urn:x-esri:serviceType:ArcGIS#ImageMapLayer'

**Pull request merged**